src/auth/src/credentials/service_account.rs

```diff
@@ -32,33 +94,117 @@ use time::OffsetDateTime;
 const DEFAULT_SCOPES: &str = "https://www.googleapis.com/auth/cloud-platform";
 
 pub(crate) fn creds_from(js: serde_json::Value) -> Result<Credential> {
-    let service_account_info =
-        serde_json::from_value::<ServiceAccountInfo>(js).map_err(CredentialError::non_retryable)?;
-    let token_provider = ServiceAccountTokenProvider {
-        service_account_info,
-    };
-    let token_provider = TokenCache::new(token_provider);
-
-    Ok(Credential {
-        inner: Arc::new(ServiceAccountCredential { token_provider }),
-    })
+    let service_account_key =
+        serde_json::from_value::<ServiceAccountKey>(js).map_err(CredentialError::non_retryable)?;
+    Ok(Builder::default()
+        .service_account_key(service_account_key)
+        .build())
+}
+
+#[derive(Default)]
+pub struct Builder {
+    service_account_key: ServiceAccountKey,
+    aud: Option<String>,
+    scopes: Option<String>,
+    quota_project_id: Option<String>,
+}
+
+impl Builder {
+    /// Sets the [service account key] in [aip/4112] format for this credential.
+    ///
+    /// [aip/4112]: https://google.aip.dev/auth/4112
+    /// [service account key]: https://cloud.google.com/iam/docs/keys-create-delete#creating
+    pub fn service_account_key(mut self, service_account_key: ServiceAccountKey) -> Self {
+        self.service_account_key = service_account_key;
+        self
+    }
```

Comment:
It seems that a builder without a service account key is useless. Should this be:

```suggestion
    /// Creates a new builder using `service_account_key` in [aip/4112] format for this credential.
    ///
    /// [aip/4112]: https://google.aip.dev/auth/4112
    /// [service account key]: https://cloud.google.com/iam/docs/keys-create-delete#creating
    pub fn new(service_account_key: ServiceAccountKey) -> Self {
        Self { service_account_key,  aud: None, scopes: None, quota_project_id: None }
    }
```